pkg/controller/volume/attachdetach/util/util_test.go

```diff
@@ -0,0 +1,316 @@
+		hostPath    = &v1.HostPathVolumeSource{
+			Path: socketPath,
+			Type: &dirOrCreate,
+		}
+		objectReference = v1.ObjectReference{Namespace: "claimrefns", Name: "test-pvc"}
+
+		volumeExampe = v1.Volume{
+			Name: volumeName,
+			VolumeSource: v1.VolumeSource{
+				HostPath: hostPath,
+			},
+		}
+		volumeExampeWithCSI = v1.Volume{
+			Name: volumeName,
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "test-pvc",
+					// TODO: Use source definition to set this value when we have one.
+					ReadOnly: false,
+				},
+			},
+		}
+		persistentVolumeExampeWithCSI = v1.PersistentVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: volumeName,
+			},
+			Spec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					CSI: &v1.CSIPersistentVolumeSource{
+						Driver:       "pd.csi.storage.gke.io",
+						VolumeHandle: "projects/UNSPECIFIED/zones/UNSPECIFIED/disks/test-disk",
+						FSType:       "ext4",
+						ReadOnly:     false,
+						VolumeAttributes: map[string]string{"partition": ""},
+					},
+				},
+				ClaimRef: &objectReference,
+			},
+		}
+	)
+
+	tmpDir, err := utiltesting.MkTmpdir("csi-test")
+	if err != nil {
+		t.Fatalf("can't make a temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	csiTranslator := csitrans.New()
+	intreeToCSITranslator := csimigration.NewPluginManager(csiTranslator, utilfeature.DefaultFeatureGate)
+	node, pod, pv, claim := createObjects(v1.PersistentVolumeFilesystem, v1.PersistentVolumeFilesystem)
+	kubeClient := fake.NewSimpleClientset(node, pod, pv, claim)
+
+	factory := informers.NewSharedInformerFactory(kubeClient, time.Minute)
+	csiDriverInformer := factory.Storage().V1().CSIDrivers()
+	csiDriverLister := csiDriverInformer.Lister()
+	volumeAttachmentInformer := factory.Storage().V1().VolumeAttachments()
+	volumeAttachmentLister := volumeAttachmentInformer.Lister()
+
+	plugMgr := &volume.VolumePluginMgr{}
+	fakeAttachDetachVolumeHost := volumetest.NewFakeAttachDetachVolumeHostWithCSINodeName(t,
+		tmpDir,
+		kubeClient,
+		fc.ProbeVolumePlugins(),
+		testNodeName,
+		csiDriverLister,
+		volumeAttachmentLister,
+	)
+
+	plugMgr.Host = fakeAttachDetachVolumeHost
+
+	pvLister := fakeframework.PersistentVolumeLister{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: volumeName},
+			Spec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+						PDName:    "test-disk",
+						FSType:    "ext4",
+						Partition: 0,
+						ReadOnly:  false,
+					},
+				},
+				ClaimRef: &objectReference,
+			},
+		},
+	}
+
+	pvcLister := fakeframework.PersistentVolumeClaimLister{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pvc", Namespace: "default"},
+			Spec:       v1.PersistentVolumeClaimSpec{VolumeName: volumeName},
+			Status: v1.PersistentVolumeClaimStatus{
+				Phase: v1.ClaimBound,
+			},
+		},
+	}
+
+	tests := []struct {
```

Comment:
It looks like the two test cases you have appear to be 1) inline volume type that does not support csi migration 2) PV type that supports csi migration.

I think it would be good to add a few more test cases:
- Inline volume type that supports csi migration
- PV type that does not support csi migration
- CSINode object does not exist for a volume type that supports csi migration